### PR TITLE
[api] Add size limit for http

### PR DESCRIPTION
**Pull request by Liuhaai**
# Description
https://github.com/ethereum/go-ethereum/blob/master/rpc/http.go#L36

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/http.go [hunk 1]
```diff
@@ -70,8 +75,8 @@ func (handler *hTTPHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 		w.WriteHeader(http.StatusMethodNotAllowed)
 		return
 	}
-
-	if err := handler.msgHandler.HandlePOSTReq(ctx, req.Body,
+	body := io.LimitReader(req.Body, _maxRequestContentLength)
```
Comment (dustinxie) [inline api/http.go:78]: no need to check error? i mean if it has > 5MB data, `body` will be incomplete, would that be an issue for the `HandlePOSTReq` call below?

Comment (millken) [reply api/http.go:78]: should be here  https://github.com/ethereum/go-ethereum/blob/master/rpc/http.go#L343
```golang
	if r.ContentLength > maxRequestContentLength {
		err := fmt.Errorf("content length too large (%d>%d)", r.ContentLength, maxRequestContentLength)
		return http.StatusRequestEntityTooLarge, err
	}
```

Comment (dustinxie) [reply api/http.go:78]: so we need a similar check to validate the request?

**Comment by raullenchai** [comment]
Close b/c it has been inactive for 1+yr.

**Pull request closed without merging**